### PR TITLE
Add logic to retry errors

### PR DESCRIPTION
**Pull request by athompson-r7**
This change add a retry decorator that can be added to requests that might fail.  The usage is something like this:

```ruby
DaemonRunner::RetryErrors.retry(exceptions: [RuntimeError]) do
  loop do
    raise RuntimeError, 'boom' if rand * 100 > 90
  end
end
```

After the default of 3 retries it will give up.

### CODE REVIEW

File: lib/daemon_runner/retry_errors.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+module DaemonRunner
+    #
+    # Retry Errors
+    #
+    class RetryErrors
+      class << self
+        def retry(retries: 3, exceptions: [Faraday::ClientError], &block)
+          Retryable.retryable({
+            on: exceptions,
```
Comment (jmanero-r7) [inline lib/daemon_runner/retry_errors.rb]: Does this need to have the `tries` property passed to it?

Comment (athompson-r7) [inline lib/daemon_runner/retry_errors.rb]: @jmanero-r7 `retries` is used in the `sleep` lambda below.

Comment (jmanero-r7) [inline lib/daemon_runner/retry_errors.rb]: What does that do? As far as I can see, you're just calculating the same power every time. Should the retry wrapper continue to do this forever, or eventually throw an exception farther up stack?

**Comment by rvanburen-r7** [comment]
LGTM

File: lib/daemon_runner/retry_errors.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            sleep: lambda { |c| Kernel.sleep(2 ** retries * 0.3) }},
```
Comment (dgreene-r7) [inline lib/daemon_runner/retry_errors.rb]: Thinking about it more you have a few options for how you want to handle this:
1. Pass the calculated retry directly to the `sleep` param.
2. Pass a lambda to the `sleep` param that does backoffs based on the number of retries.
3. Pass the lambda you wrote to the `sleep_method` param.

File: lib/daemon_runner/retry_errors.rb (hunk 1 above)
Comment (dgreene-r7) [inline lib/daemon_runner/retry_errors.rb]: The default number of `tries` is [2](https://github.com/nfedyashev/retryable/blob/master/lib/retryable/configuration.rb#L34) so it'll only sleep the calculated number of seconds twice before throwing.

**Comment by athompson-r7** [comment]
The sleep seems to be working like intended now. @dgreene-r7 @jmanero-r7 What to take another look?

```
W, [2016-10-25T12:55:35.800683 #92849]  WARN : try 1 failed with exception: ArgumentError
W, [2016-10-25T12:55:36.401319 #92849]  WARN : try 2 failed with exception: ArgumentError
W, [2016-10-25T12:55:37.605800 #92849]  WARN : try 3 failed with exception: ArgumentError
W, [2016-10-25T12:55:40.009851 #92849]  WARN : try 4 failed with exception: ArgumentError
W, [2016-10-25T12:55:44.814049 #92849]  WARN : try 5 failed with exception: ArgumentError
W, [2016-10-25T12:55:54.416765 #92849]  WARN : try 6 failed with exception: ArgumentError
bundler: failed to load command: ./example_retry_errors.rb (./example_retry_errors.rb)
ArgumentError: ArgumentError
```

**Comment by dgreene-r7** [comment]
Wanna write a few tests?

File: lib/daemon_runner/retry_errors.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+module DaemonRunner
+    #
+    # Retry Errors
+    #
+    class RetryErrors
+      extend Logger
+
+      class << self
+        def retry(retries: 3, exceptions: [Faraday::ClientError], &block)
+          Retryable.retryable({
```
Comment (jmanero-r7) [inline lib/daemon_runner/retry_errors.rb]: STYLE: The `{}` should be unnecessary here. You should be able define the properties on one line with the opening `do` of the code block and make the block a lot more readable.

**Comment by athompson-r7** [comment]
> Wanna write a few tests?

@dgreene-r7 Added some tests for this in 	35cc958.

File: lib/daemon_runner/retry_errors.rb (hunk 3 above)
Comment (athompson-r7) [inline lib/daemon_runner/retry_errors.rb]: @jmanero-r7 Addressed in be68bb6.

**Pull request merged**